### ISSUE TITLE
Model->exists() inconsistent behavior with $id = 0 when called without $id parameter

### ISSUE THREAD

**Issue by lorenzomaieru** [cakephp/cakephp#7627]
*This is probably and edge case, considering that the premise could be considered a bad practice, but:*

In CakePHP 2.x, assuming we have a row in our database with id = 0,

This piece of code returns false
```php
$id = 0;
$this->Model->id = $id;
return $this->Model->exists(); //returns false
```

While this piece of code returns true
```php
$id = 0;
return $this->Model->exists($id); //returns true
```

this seems to be caused by the [call to Model->getID](https://github.com/cakephp/cakephp/blob/2.7/lib/Cake/Model/Model.php#L2895-L2897) in the Model->exists method:
```php
public function exists($id = null) {
		if ($id === null) {
			$id = $this->getID();
		}
```

the Model->getID method has an [empty() check](https://github.com/cakephp/cakephp/blob/2.7/lib/Cake/Model/Model.php#L3546-L3549) on the $id property of the model,
```php
public function getID($list = 0) {
		if (empty($this->id) || (is_array($this->id) && isset($this->id[0]) && empty($this->id[0]))) {
			return false;
		}
```
which will return true for the value = 0

**In conclusion:** if the parameter is passed through to the exists method, getID won't get called and no empty check will be performed on the ID value, resulting in two different behaviors for pieces of code that, I would assume, should be functionally identical.

One possible fix is to replace `empty($this->id)` with `($this->id === null)`
or, going the other way, adding an empty check at the beginning of the exists method.

**Comment by dereuromark** [comment]
Duplicate of another ticket which I cannot find right now.
Basically it boils down to:
0 is an invalid start point for an auto increment ID field (mininum 1) and as such this ticket is invalid.

**Comment by markstory** [comment]
Its totally an edge case and something that will probably never be fixed, as it is extremely rare to use ID=0
